### PR TITLE
adding benchmark tests for detect funcs

### PR DESCRIPTION
**Pull request by WheeskyJack**
Hi @gabriel-vasile ,
This PR adds the golang benchmark tests for detect(), detetcreader() , and detectfile() functions

The test cases are run using 
go test -benchmem -timeout 20m -run=^$ -bench=. 

Regards.

### CODE REVIEW

**Comment by gabriel-vasile** [comment]
Hi @WheeskyJack, I'm thinking of keeping just `BenchmarkDetect`. I don't see need for three benchmarks because in the background all three do the same thing.

File: benchmark_test.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var testresult string
```
[the remark this answers is not in the document]
Comment (WheeskyJack) [reply benchmark_test.go]: Hi,
Thanks for reviewing!
The scope of variable is intentional here.
It then doesn't allow compiler to do optimization and hence more accurate results for timing are possible.
Reference :   
https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go

Regards.

**Comment by WheeskyJack** [comment]
Hi @gabriel-vasile ,
I agree with you on keeping benchmark for only one function.
The thinking behind writing for all the three function was that , those three functions are exposed to the outside world and we should always ideally have tests for our external interfaces. So just to cover all our bases, those benchmarks were written. 
Note that those tests will benchmark file read and io read operations for file and reader functions respectively.
Sometime these operations tends to be more costly than actual code logic. 
And also, we can execute specific benchmark test in go test command; so if executing all benchmarks takes more time, we can always execute specific benchmarks.
If you still feel those other benchmarks are redundant , please comment and we can remove them.
Thanks for review! :)
Regards.

File: benchmark_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabriel-vasile) [reply benchmark_test.go]: Hello, nice catch with that variable, guess we'll leave it there.

**Comment by gabriel-vasile** [comment]
The core functionality of the library is in the `Detect` function, other versions of `Detect` are just convenience functions. Benchmarking reader and file detect only tells info about golang's stdlib performance. 
Also the benchmark is run on a lot of files and more to be added. We should stick with just one benchmark.

**Comment by gabriel-vasile** [comment]
Closing this due to inactivity.

**Pull request closed without merging**